src/exportJarSteps/GenerateJarExecutor.ts

```diff
@@ -91,22 +91,28 @@ export class GenerateJarExecutor implements IExportJarStepExecutor {
                 });
                 const pickItems: IJarQuickPickItem[] = [];
                 const uriSet: Set<string> = new Set<string>();
-                for (const rootNode of stepMetadata.projectList) {
-                    const classPaths: ClasspathResult = await extensionApi.getClasspaths(rootNode.uri, { scope: "runtime" });
-                    pickItems.push(...await this.parseDependencyItems(classPaths.classpaths, uriSet, stepMetadata.workspaceFolder.uri.fsPath, true),
-                        ...await this.parseDependencyItems(classPaths.modulepaths, uriSet, stepMetadata.workspaceFolder.uri.fsPath, true));
-                    const classPathsTest: ClasspathResult = await extensionApi.getClasspaths(rootNode.uri, { scope: "test" });
-                    pickItems.push(...await this.parseDependencyItems(classPathsTest.classpaths, uriSet,
-                        stepMetadata.workspaceFolder.uri.fsPath, false),
-                        ...await this.parseDependencyItems(classPathsTest.modulepaths, uriSet, stepMetadata.workspaceFolder.uri.fsPath, false));
+                for (const project of stepMetadata.projectList) {
+                    const runTimeClassPaths: IClasspathResult = await extensionApi.getClasspaths(project.uri, { scope: "runtime" });
+                    pickItems.push(
+                        ...await this.parseDependencyItems(runTimeClassPaths.classpaths, uriSet, stepMetadata.workspaceFolder.uri.fsPath, true),
+                        ...await this.parseDependencyItems(runTimeClassPaths.modulepaths, uriSet, stepMetadata.workspaceFolder.uri.fsPath, true));
+                    const testClassPaths: IClasspathResult = await extensionApi.getClasspaths(project.uri, { scope: "test" });
+                    pickItems.push(
+                        ...await this.parseDependencyItems(testClassPaths.classpaths, uriSet, stepMetadata.workspaceFolder.uri.fsPath, false),
+                        ...await this.parseDependencyItems(testClassPaths.modulepaths, uriSet, stepMetadata.workspaceFolder.uri.fsPath, false));
                 }
                 return resolve(pickItems);
             });
         });
         if (_.isEmpty(dependencyItems)) {
             throw new Error("No classpath found. Please make sure your java project is valid.");
         } else if (dependencyItems.length === 1) {
-            stepMetadata.elements.push(dependencyItems[0].path);
+            const classpath: IClassPaths = {
+                source: dependencyItems[0].path,
+                destination: undefined,
+                isExtract: false,
```

Comment:
What's the meanning of `isExtract`

Reply:
If we want to extract the sources rather than just copy them into the jar file, for example, the dependencies, we will set this field to true. Otherwise false.

Reply:
Ok, so is this a user visible field?

Reply:
No. We will set this field to `true` if an `IClassPaths` was generated from dependencies. But I will also change this name to a more readable one in the next commit.